pennylane/transforms/metric_tensor.py

```diff
@@ -606,7 +606,10 @@ def processing_fn(results):
         # Initialize off block-diagonal tensor using the stored ids
         first_term = qml.math.zeros_like(diag_mt)
         if ids:
-            off_diag_res = qml.math.stack(off_diag_res, 1)[0]
+            if qml.math.ndim(off_diag_res[0]) > 0:
+                off_diag_res = qml.math.stack(off_diag_res, 1)[0]
+            else:
+                off_diag_res = qml.math.stack(off_diag_res)

```

Comment:
The off-diagonal entries are computing with the special tapes created in `_get_first_term_tapes`, which all have `expval(PauliX(aux_wire))` as measurement. I'm therefore surprised that we need to cover the case in which one of the `off_diag_res` is not a scalar. 
After running the tests I see _that_ it happens but don't get yet _why_ :thinking: 

**edit** The decision whether to squeeze the `off_diag_res` results is made based on the QNode output shape, but for these tapes, the QNode output does not matter, it always is a single expectation value.
Therefore, squeezing should always but doesn't always happen, as far as i can tell.

Reply:
In https://github.com/PennyLaneAI/adrs/blob/master/documents/052-multiple-circuits.md we propose that we modify the QNode to contain/construct multiple circuits/tapes 🙂 

So once implemented, the `batch_transform` will simply return a new QNode that contains multiple tapes, rather than the current approach where it 'intercepts' the QNode between construction and execution.

This should make everything cleaner!

Reply:
To prevent this and other issues with transforms that use the batch transform differently than the gradient transforms (e.g. the new qcut transform was also raising errors) I've moved the squeezing fix to inside the gradient transform's postprocessing functions, instead of at the end of the batch transform.